src/main/kotlin/org/rust/ide/inspections/SelfConventionInspection.kt

```diff
@@ -15,7 +15,7 @@ class SelfConventionInspection : RustLocalInspectionTool() {
             override fun visitImplMethodMember(m: RustImplMethodMember) {
                 val convention = SELF_CONVENTIONS.find { m.identifier.text.startsWith(it.prefix) } ?: return
                 if (m.selfType !in convention.selfTypes) {
-                    holder.registerProblem(m.selfArgument ?: m.identifier, convention)
+                    holder.registerProblem(m.fnParamsWithSelf.selfArgument ?: m.identifier, convention)
```

Comment:
As far as I understand it is impossible to remove particular grammar kit accessor.  For this particular case, I think we should use something like

```
fake argument ::= pat? type
fake params ::= Self? argument*

trait_fn_argument ::= pat? type {
  elementType = argument
}
trait_fn_params ::= ... {
   elementType = params
}

impl_fn_argument ::= pat type {
   elementType = argument
}
impl_fn_params ::= ... {
  elementType = params
}
```

This should generate `params` accessor everywhere. And I think we should not have different PSI nodes for params with and without self.